### DISCUSSION TITLE
请教大家：我部署到github的网页加载js失败，原因为由于严格的MIME检测策略

### DISCUSSION THREAD

**Discussion by Tru-tru** [YunYouJun/hexo-theme-starry#335] [Q&A 疑难杂症] [answered]
我的网页：https://tru-tru.github.io/

错误日志：
```code
Error with Permissions-Policy header: Origin trial controlled feature not enabled: 'interest-cohort'.
tru-tru.github.io/:1 Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/html". Strict MIME type checking is enforced for module scripts per HTML spec.
tru-tru.github.io/:1 Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/html". Strict MIME type checking is enforced for module scripts per HTML spec.
tru-tru.github.io/:1 Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/html". Strict MIME type checking is enforced for module scripts per HTML spec.
tru-tru.github.io/:1 Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/html". Strict MIME type checking is enforced for module scripts per HTML spec.
tru-tru.github.io/:1 Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/html". Strict MIME type checking is enforced for module scripts per HTML spec.
```

我的配置如下（均为最新版本）：
![image](https://user-images.githubusercontent.com/108524337/184551559-4706f899-64a1-4817-8826-78669aadc1f1.png)
浏览器是：Google Chrome最新版本 版本 104.0.5112.81（正式版本） （64 位）

<strong>请问大家该如何解决这个问题？</strong>

---

我查询的资料有：

VUE解决方法：[Why is “strict MIME type checking” blocking the static serving of Vue frontend files?](https://medium.com/developer-rants/why-is-strict-mime-type-checking-blocking-the-static-serving-of-vue-frontend-files-4cbea1eedbd1)

还有可能的修复策略：[How To Fix Failed To Load Module Script](https://www.dev2qa.com/how-to-fix-failed-to-load-module-script-expected-a-javascript-module-script-but-the-server-responded-with-a-mime-type-of-text-plain-strict-mime-type-checking-is-enforced-for-module-scripts-per-h/)

**Comment by YunYouJun** [comment, marked as answer]
页面内容看起来经过了奇怪的转译：`<html lang="[&quot;zh-CN&quot;,&quot;default&quot;]">`

检查是否有安装奇怪的插件，最好可以提供仓库源码地址。

**Reply by Tru-tru** [reply]
你好，我的仓库地址是：https://github.com/Tru-tru/tru-tru.github.io

**Reply by YunYouJun** [reply]
<img width="1625" alt="image" src="https://user-images.githubusercontent.com/25154432/185792496-0cf13572-be74-421b-b32a-48b49d749b6e.png">

您的仓库无法查看。

**Reply by Tru-tru** [reply]
感谢你的关注，这个问题已经解决，
原因是:我在 ``_config.yml`` 文件中配置的个人网站链接为 https://Tru-tru.github.io/tru-tru.github.io

修正为 https://tru-tru.github.io/ 即解决此问题

---
<details><summary>错误原因</summary>
<br>
我看英文介绍

``` yml
## Set your site url here. For example, if you use GitHub Page, set url as 'https://username.github.io/project'
```
理解错了此项的配置；

而https://github.com/YunYouJun/yunyoujun.github.io/blob/hexo/_config.yml 中已经指明
```
# # If your site is put in a subdirectory, set url as 'https://yoursite.com/child' and root as '/child/'
```
当博客使用子目录时才需要配置成https://Tru-tru.github.io/tru-tru.github.io 形式
</details>

**Reply by YunYouJun** [reply]
```yml
## Set your site url here. For example, if you use GitHub Page, set url as 'https://username.github.io/project'
```

这部分出现在哪里呢？是 Hexo 的文档吗？

**Reply by Tru-tru** [reply]
是在 [hexo-stater](https://github.com/hexojs/hexo-starter) 的`config.yml`文件中：

https://github.com/hexojs/hexo-starter/blob/764892aef6df75c63e31a5458e141d5eca26aaf4/_config.yml#L14-L15

**Reply by YunYouJun** [reply]
那就是 Hexo 的说明不够详细了，我这边就不做什么修改了。
